sdk/attestation/Azure.Security.Attestation/src/AttestationAdministrationClient.cs

```diff
@@ -219,35 +221,24 @@ public virtual async Task<AttestationResponse<string>> GetPolicyAsync(Attestatio
         public virtual AttestationResponse<PolicyResult> SetPolicy(
             AttestationType attestationType,
             string policyToSet,
-            AsymmetricAlgorithm signingKey = default,
-            X509Certificate2 signingCertificate = default,
+            TokenSigningKey signingKey = default,
             CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrWhiteSpace(policyToSet, nameof(policyToSet));
 
-            if (signingKey is null && signingCertificate is not null || signingCertificate is null && signingKey is not null)
-            {
-                throw new ArgumentException($"If you specify '{nameof(signingKey)}' or '{nameof(signingCertificate)}', you must also specify '{nameof(signingCertificate)}' or '{nameof(signingKey)}'.");
-            }
-
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(AttestationAdministrationClient)}.{nameof(SetPolicy)}");
             scope.Start();
             try
             {
-                AttestationToken tokenToSet;
-                if (signingKey is null)
-                {
-                    tokenToSet = new UnsecuredAttestationToken(new StoredAttestationPolicy { AttestationPolicy = policyToSet, });
-                }
-                else
-                {
-                    tokenToSet = new SecuredAttestationToken(new StoredAttestationPolicy { AttestationPolicy = policyToSet,}, signingKey, signingCertificate);
-                }
+                AttestationToken tokenToSet = new AttestationToken(
+                    new StoredAttestationPolicy { AttestationPolicy = policyToSet, },
+                    signingKey);
+
                 var result = _policyClient.Set(attestationType, tokenToSet.ToString(), cancellationToken);
                 var token = new AttestationToken(result.Value.Token);
-                if (_options.ValidateAttestationTokens)
+                if (_options.TokenOptions.ValidateToken)
                 {
-                    token.ValidateToken(GetSigners(), _options.ValidationCallback);
+                    token.ValidateToken(_options.TokenOptions, GetSigners(), cancellationToken).Wait(cancellationToken);
```

Comment:
The reason that the ValidateToken API is async is because at some point, I might want to add referencing the target of the JKU header property, which will be async. But I would rather not have to build a sync and async version of ValidateToken if at all possible.

Reply:
Building the async and sync version :(. It's the right thing to do for a public API.